src/execution/riscv64/simulator-riscv64.h

```diff
@@ -497,10 +498,12 @@ class Simulator : public SimulatorBase {
     set_register(rvc_rd_reg(), value);
     if (trace) TraceRegWr(get_register(rvc_rd_reg()), DWORD);
   }
-  inline int16_t shamt() const { return (imm12() & 0x3F); }
-  inline int16_t shamt32() const { return (imm12() & 0x1F); }
+  inline int16_t shamt6() const { return (imm12() & 0x3F); }
+  inline int16_t shamt5() const { return (imm12() & 0x1F); }
+  inline int16_t rvc_shamt6() const { return instr_.RvcShamt6(); }
   inline int32_t s_imm12() const { return instr_.StoreOffset(); }
-  inline int32_t u_imm() const { return instr_.Imm20UValue() << 12; }
+  inline int32_t u_imm20() const { return instr_.Imm20UValue() << 12; }
+  inline int32_t rvc_u_imm20() const { return instr_.RvcImm6UValue() << 12; }
```

Comment:
This one should shift left by `20-6 = 14`, right?

Reply:
The number is 12 here because the lowest bit means the bit is in position 12. So it should be shifted left 12.

![图片](https://user-images.githubusercontent.com/17661043/93833630-1d0e2300-fcb4-11ea-93b3-f88259637a32.png)

Besides, we should change the function name `rvc_u_imm20` -> `rvc_u_imm6`, right?

Reply:
hmm, `rvs_u_imm6` is kind of confusing too. Since one is not obviously better than the other, let's keep the `u_imm20` and `rvc_u_imm20` naming because we use `Imm20UValue()`.

Reply:
`rvs_u_imm6` is consistent with `RvcImm6Value()`, just like `rvc_u_imm20` and `Imm20UValue()`, which means the number of immediate bits in the instruction. 

I could not see what the meaning of `20` in `rvc_u_imm20`. I thought the number reflects the bits in raw instruction, but not the value.